packages/data-context/src/sources/HtmlDataSource.ts

```diff
@@ -41,8 +42,32 @@ export class HtmlDataSource {
     throw err
   }
 
+  getUrlsFromLegacyProjectBase (cfg: any) {
```

Comment:
The config we get in the data-context doesn't have some of these important keys. They are dynamically added later, for example the `port` will only be added once the CT dev server has started.

We'll need to find a way to better manage the config in general, but this fixes the problem for now and links the "legacy" part of config in `project-base` and `server-{ct,e2e}` with the newer data-context stuff.